pkg/synccompactor/compactor.go

```diff
@@ -0,0 +1,124 @@
+package synccompactor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+
+	reader_v2 "github.com/conductorone/baton-sdk/pb/c1/reader/v2"
+	"github.com/conductorone/baton-sdk/pkg/dotc1z"
+	c1zmanager "github.com/conductorone/baton-sdk/pkg/dotc1z/manager"
+	sync_compactor "github.com/conductorone/baton-sdk/pkg/synccompactor/naive"
+)
+
+type Compactor struct {
+	entries []*CompactableSync
+	destDir string
+}
+
+type CompactableSync struct {
+	filePath string
+	syncID   string
+}
+
+func NewCompactor(ctx context.Context, destDir string, compactableSyncs ...*CompactableSync) (*Compactor, error) {
+	if len(compactableSyncs) == 0 || len(compactableSyncs) == 1 {
+		return nil, errors.New("must provide two or more files to Compact")
+	}
+
+	return &Compactor{entries: compactableSyncs, destDir: destDir}, nil
+}
+
+func (c *Compactor) Compact(ctx context.Context) (*CompactableSync, error) {
+	if len(c.entries) < 2 {
+		return nil, nil
+	}
+
+	tempDir := os.TempDir()
```

Comment:
It's fine to call NewC1ZFile() without a temp dir, as it calls os.MkdirTemp() which, if the first argument is empty, uses os.TempDir().

Reply:
great

Reply:
But how do I get the filename?

Reply:
Just kidding, I still pass the filename as-is I just remove the temp dir